apps/web/src/hooks/useLatitudeAction.ts

```diff
@@ -0,0 +1,54 @@
+import { useCallback } from 'react'
+
+import {
+  inferServerActionError,
+  inferServerActionReturnData,
+  TAnyZodSafeFunctionHandler,
+} from 'zsa'
+
+function formDataToJson(data: FormData) {
+  const json: { [key: string]: unknown } = {}
+  data.forEach((value, key) => {
+    json[key] = value
+  })
+  return json
+}
+
+export default function useLatitudeAction<
+  const TServerAction extends TAnyZodSafeFunctionHandler,
+>(
+  action: TServerAction,
+  {
+    onSuccess,
+    onError,
+  }: {
+    onSuccess: (payload: inferServerActionReturnData<TServerAction>) => void
+    onError: (error: inferServerActionError<TServerAction>) => void
+  } = { onSuccess: () => {}, onError: () => {} },
+) {
+  const execute = useCallback(
+    async (data: { [key: string]: unknown }) => {
+      const [payload, error] = await action(data)
+
+      if (error) {
+        return onError(error)
+      }
+      return onSuccess(payload)
+    },
+    [action, onSuccess, onError],
+  )
+
+  const executeFormAction = useCallback(
+    async (data: FormData) => {
+      const [payload, error] = await action(formDataToJson(data))
+
+      if (error) {
+        return onError(error)
+      }
+      return onSuccess(payload)
+    },
+    [action, onSuccess, onError],
+  )
+
+  return { execute, executeFormAction }
+}
```

Comment:
simple now you can use actions both in forms and as regular imperative methods without having to modify the signature of actions (which should always accept json)